storage/scratchpad/src/sparse_merkle/node.rs

```diff
@@ -76,26 +78,58 @@ where
 }
 
 #[derive(Debug)]
-pub(crate) enum Node<V> {
+pub(crate) enum NodeInner<V> {
     Internal(InternalNode<V>),
     Leaf(LeafNode<V>),
 }
 
+#[derive(Debug)]
+pub(crate) struct Node<V> {
+    generation: u64,
```

Comment:
what is this for?

Reply:
Even if we hold a reference to the oldest ancestor, there still can be nodes that's before it BEING dropped from another thread (the root has been dropped, but the nodes under it is BEING dropped.)